roachprod: fix unattended service on startup

herkolategan opened the pull request:
Previously on GCE, the unattended-upgrade service was stopped and removed before
the OS initialisation check. On a managed cluster where the VM can be restarted
this would cause start-up to fail since it tries to stop a non-existent service.
This change moves the stop/remove to after the OS initialisation check.

Fixes: https://github.com/cockroachdb/cockroach/issues/147044

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/147017)
<!-- Reviewable:end -->

srosenberg reviewed (approved):
Thanks, LGTM! Let's wait for @golgeek to stamp.

We'll need to create a smoke test to restart VMs and make sure startup scripts work as expected, in the corresponding clouds. @DarrylWong Just noting that it's also something FI would easily pick up, when we start injecting VM restarts.

herkolategan commented on pkg/roachprod/vm/gce/testdata/startup_script:
```diff
@@ -115,10 +115,6 @@ if [ -e /mnt/data1/.roachprod-initialized ]; then
 	exit 0
 fi
 
-# Uninstall some packages to prevent them running cronjobs and similar jobs in parallel
```
replace with
```
if systemctl list-unit-files | grep -q '^unattended-upgrades.service'; then
    echo "unattended-upgrades service exists. Proceeding with uninstallation."
    systemctl stop unattended-upgrades
    sudo rm -rf /var/log/unattended-upgrades
    apt-get purge -y unattended-upgrades
fi
```
to keep as early as possible

herkolategan committed:
roachprod: fix unattended service on startup

Previously on GCE, the unattended-upgrade service was stopped and removed before
the OS initialisation check. On a managed cluster where the VM can be restarted
this would cause start-up to fail since it tries to stop a non-existent service.

This should still be kept as early on as possible, so we rather check if the
service has already been removed before attempting to do so.

Fixes: #147044

Epic: None
Release note: None

golgeek reviewed (approved):
LGTM!
